core/src/main/scala/org/ensime/indexer/SourceResolver.scala

```diff
@@ -19,12 +21,15 @@ class SourceResolver(
     vfs: EnsimeVFS
 ) extends FileChangeListener with SLF4JLogging {
 
-  // it's not worth doing incremental updates - this is cheap
-  // (but it would be nice to have a "debounce" throttler)
-  def fileAdded(f: FileObject) = update()
-  def fileRemoved(f: FileObject) = update()
+  def fileAdded(f: FileObject) = if (relevant(f)) update()
+  def fileRemoved(f: FileObject) = if (relevant(f)) update()
   def fileChanged(f: FileObject) = {}
 
+  def relevant(f: FileObject): Boolean = f.getName.isFile && {
+    val file = new File(f.getName.getURI)
```

Comment:
@fommil  says 'It's only called for changed files so it's not a churn hotspot'
LGTM